x/consensus/module.go

```diff
@@ -0,0 +1,172 @@
+	return nil
+}
+
+// RegisterInterfaces registers interfaces and implementations of the bank module.
+func (AppModuleBasic) RegisterInterfaces(registry codectypes.InterfaceRegistry) {
+	types.RegisterInterfaces(registry)
+}
+
+// AppModule implements an application module
+type AppModule struct {
+	AppModuleBasic
+
+	keeper keeper.Keeper
+
+	// legacySubspace is used solely for migration of x/params managed parameters
+	legacySubspace exported.ParamStore
+}
+
+// RegisterServices registers module services.
+func (am AppModule) RegisterServices(cfg module.Configurator) {
+	types.RegisterMsgServer(cfg.MsgServer(), keeper.NewMsgServerImpl(am.keeper))
+	types.RegisterQueryServer(cfg.QueryServer(), keeper.NewQuerier(am.keeper))
+}
+
+// NewAppModule creates a new AppModule object
+func NewAppModule(cdc codec.Codec, keeper keeper.Keeper, ss exported.ParamStore) AppModule {
+	return AppModule{
+		AppModuleBasic: AppModuleBasic{cdc: cdc},
+		keeper:         keeper,
+		legacySubspace: ss,
+	}
+}
+
+// Name returns the consensus_param module's name.
+func (AppModule) Name() string { return types.ModuleName }
+
+// InitGenesis is handled by for init genesis of consensus_param
+func (am AppModule) InitGenesis(ctx sdk.Context, cdc codec.JSONCodec, data json.RawMessage) []abci.ValidatorUpdate {
+	return nil
+}
+
+// ExportGenesis is handled by tendermint export of genesis
+func (am AppModule) ExportGenesis(ctx sdk.Context, cdc codec.JSONCodec) json.RawMessage { return nil }
+
+// ConsensusVersion implements AppModule/ConsensusVersion.
+func (AppModule) ConsensusVersion() uint64 { return ConsensusVersion }
+
+// RegisterInvariants does nothing, there are no invariants to enforce
+func (am AppModule) RegisterInvariants(ir sdk.InvariantRegistry) {}
+
+func init() {
+	appmodule.Register(
+		&modulev1.Module{},
+		appmodule.Provide(provideModuleBasic, provideModule),
+	)
+}
+
+func provideModuleBasic() runtime.AppModuleBasicWrapper {
+	return runtime.WrapAppModuleBasic(AppModuleBasic{})
+}
+
+type consensusParamInputs struct {
+	depinject.In
+
+	Cdc       codec.Codec
+	Key       *store.KVStoreKey
+	ModuleKey depinject.OwnModuleKey
+	Authority map[string]sdk.AccAddress `optional:"true"`
+
+	// LegacySubspace is used solely for migration of x/params managed parameters
+	LegacySubspace exported.ParamStore
+}
+
+type consensusParamOutputs struct {
+	depinject.Out
+
+	Keeper        keeper.Keeper
+	Module        runtime.AppModuleWrapper
+	BaseAppOption runtime.BaseAppOption
+}
+
+func provideModule(in consensusParamInputs) consensusParamOutputs {
+	authority, ok := in.Authority[depinject.ModuleKey(in.ModuleKey).Name()]
+	if !ok {
+		// default to governance authority if not provided
+		authority = authtypes.NewModuleAddress(govtypes.ModuleName)
+	}
+
+	k := keeper.NewKeeper(in.Cdc, in.Key, authority.String())
+	m := NewAppModule(in.Cdc, k, in.LegacySubspace)
+	// baseappOpt := func(app *baseapp.BaseApp) {
+	// 	app.SetParamStore(&k)
+	// }
```

Comment:
is this gonna be re-implemented?

Reply:
yea in the pr that handles the update to BaseApp. the interface changes